src/data/mutable_data.rs

```diff
@@ -395,6 +395,42 @@ impl MutableData {
         Ok(())
     }
 
+    /// Mutates entries without performing any validation.
+    ///
+    /// For updates and deletes, the mutation is performed only if he entry version
+    /// of the action is higher than the current version of the entry.
+    pub fn mutate_entries_without_validation(&mut self, actions: BTreeMap<Vec<u8>, EntryAction>) {
```

Comment:
`mutate_entries` is still used in vaults, but actually only to perform the validation before inserting it into pending writes. The mutation itself is thrown away at that point. So yeah, it's bit of a mess :grimacing: 

I think we should have three methods:
1. `validate_mutate_entries(&self, actions: ...) -> Result<(), ClientError>` - only validate, no mutation
2. `mutate_entries_without_validation(&mut self, actions: ...)` - no validation, only mutation
3. `mutate_entries(&mut self, actions: ...)` - do both, but by calling the above two
(the names are up for discussion)